### DISCUSSION TITLE
Zigbee2Tasmota (Z2T) on sonoff bridge

### DISCUSSION THREAD

**Discussion by bartplessers** [arendst/Tasmota#14911] [Q&A] [answered]
Hi,

I use Zigbee2mqtt + mqtt broker + Home Assistant mqtt integration for some years now. Works very well.

Now I want to have an "independent" z2m bridge that runs on its own hardware.

I can use a dedicated RPI for this, but this is to much overhead:
- the OS needs regular maintenance 
- risk of sd-card corruption
- zigbee stick needs usb extention cable to avoid interference
- power supply needed of 3A for RPI4
- hard to find RPI's these days
- price for RPI, power supply, SD-card. ...

So I was thinking of using Z2T on a sonoff zigbee bridge.

Some questions:
- can I use Z2T for this purpose? Thus: can I pair all my devices, create a zigbee mesh and read/set properties only by using mqtt messages on my broker?
- what about Home Assistant integration? Does HA autodiscovers the devices/entities on the mqtt broker? Is the mqtt structure of the published messages compatible with autodiscover?
- Does Z2T also has some advanced features like OTA updates of the paired devices
- does Z2T supports as many devices as Z2M
- is sonoff zigbee bridge as powerfull as my current zhh stick?
- can I create different zigbee networks for my different appartement (with different sonoff bridges ofcourse) and connect them to the same mqtt-broker?

Thanx in advance!
Bart

**Comment by sfromis** [comment, marked as answer]
Z2T is similar to Z2M, in that it works via MQTT payloads. However it is not a compatible replacement, and Z2M may have more automatic device handling built-in. With Z2T, there is no HA auto discovery, and no OTA updates of paired devices. Due to the low capacity of the ESP8266 in Sonoff ZbBridge, not all features from Z2M could be implemented. No problem having multiple Z2T devices on the same broker. There also exists a ESP32-based bridge very similar to the Sonoff, which allows for future enhancements over what's possible with ESP8266.

If you do all your backend stuff via HA, another option would be to run in ZHA mode (without any MQTT), allowing ZHA to take control over the Zigbee chip in the bridge (via ser2net), for more smooth integration.

**Reply by bartplessers** [reply]
Thanx for this clear answer!
I would like to stick on my central MQTT broker as 'central point of truth', so ZHA is not an option. I have several instances of HA (dev/quality/production) in sync, all connected to same broker. Some instances of domoticz are also connected (for long term graphs). Even some OpenHab installations running on same broker...

I do have some spare parts (esp8266, esp32 and cc2531), so I will start to experiment with this first. Any good guidance to setup the bridge with those components?

No HA autodiscovery is a bummer and this surprises me. Will it be considered in the future? I think you only have to be compatible with a certain HA-compatible mqtt structure? Or am I wrong on this?

Anyway, thanx for your clearification!
Kind regards,
B

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/CC2530/

What "will be considered" is not some sort of planning, more about who decides on doing something, and submit a PR. That's open source.... With Berry on ESP32, it should be reasonably feasible (without going very deep), to create payloads adapted for HA or whatever, without this "needing" to be in core Tsamota.

**Reply by s-hadinger** [reply]
The main difficulty with autodiscovery is from the Zigbee protocol itself. There is no real announce from devices to say what they are. Tasmota does some sort of cluster discovery to guess what category of device it is. For example, cluster 0x0006 just tells you that the device can be switched on/off. Cluster 0x0300 tells that it's a bulb, but you don't know what kind of bulb it is (white with color temp, RGB...). We can certainly do better than nothing but don't expect miracles.

Z2M compensates with an extensive database of all Zigbee devices and their features. This is very high maintenance and requires resources that an ESP doesn't have.

**Reply by bartplessers** [reply]
>Z2M compensates with an extensive database of all Zigbee devices and their features. This is very high maintenance and requires resources that an ESP doesn't have.
I certainly can understand that.

When a zigbee device announces itself, doesn't is send a "model id" or something where you can deduce it's capacities? Maybe the Z2M community would allow to access their database so with some rest calls you could find the specifications of the device. Well I will keep on dreaming I supppose... ;-)

So probably I will stick on a dedicated RPI + Z2M, eventually with a MQTT broker already on it and consider it as an appliance...

Just, If you think about an Ikea, Philps, Tuya, ... - hub: they just do exactly the same: act as a coordinator, handle all the zigbee requests and open an API (although not with MQTT) to bridge it to the normal network. It would be great to have an open zigbee hub with less hardware needed.


kind regards,
Bart

**Reply by s-hadinger** [reply]
> When a zigbee device announces itself, doesn't is send a "model id" or something where you can deduce it's capacities?

Indeed devices send a "model id", for ex: `TS0501A` and manufacturer `_TZ3000_yeg1e5eh`. It's hard to deduce anything from it. And Z2M doesn't have an actual database, it is javascript code created for each device. This is really high maintenance and frequent upgrades.

> Just, If you think about an Ikea, Philps, Tuya, ... - hub: they just do exactly the same: act as a coordinator, handle all the zigbee requests and open an API (although not with MQTT) to bridge it to the normal network. It would be great to have an open zigbee hub with less hardware needed.

"open zigbee hub" with an API. It looks very close to Z2T per your definition, no?

**Reply by sfromis** [reply]
Technically, it would seem doable to create a Berry system to automate, at least part of the way, via accessing zigbee2mqtt.io data like:
https://github.com/Koenkk/zigbee2mqtt.io/tree/master/docs/supported-devices
https://github.com/Koenkk/zigbee2mqtt.io/tree/master/docs/devices

**Reply by bartplessers** [reply]
> "open zigbee hub" with an API. It looks very close to Z2T per your definition, no?

😁

True. If it could be packaged in less demanding hardware...

And as I said: probably I will stick on that setup too
I'm just big fan of tasmota and have quite a lot of devices running on it.
So having a tasmota zigbee bridge would have been the crown on that work 😉
